src/core/memory.cc

```diff
@@ -1398,12 +1541,13 @@ void* allocate(size_t size) {
     if (size <= sizeof(free_object)) {
         size = sizeof(free_object);
     }
+    size_t sample_weight = get_cpu_mem().should_sample(size);
```

Comment:
What does `should_sample` mean in the context of `free`?

As I understand it, we need to "handle sampling" is in `free` is entirely dependent on whether the allocation is a sampled one or not, which is decided at allocation time. That we, it doesn't make sense to separate decide to sample alloc or free: each allocated piece of memory is either sampled (needing handling at alloc + free) or not (no special handling for either).

Reply:
You are definitely correct. Not entirely sure I understand the location of the comment though as this is in `allocate` and not `free`. 

I am wondering whether github tricked you here with the diff view? 

<img width="655" alt="image" src="https://user-images.githubusercontent.com/2904723/235702097-0f868c9f-6e9d-44ca-a293-254fb7a34b9a.png">

Reply:
@StephanDollberg - haha yes, that's almost certainly what happened and perhaps also the mention of `free_object` also above.